api/admin/controller.py

```diff
@@ -955,6 +955,58 @@ def _count_complaints_for_work(self, work):
         complaint_types = [complaint.type for complaint in work.complaints if not complaint.resolved]
         return Counter(complaint_types)
 
+    def custom_lists(self, identifier_type, identifier):
+        library = flask.request.library
+        work = self.load_work(library, identifier_type, identifier)
+        if isinstance(work, ProblemDetail):
+            return work
+
+        staff_data_source = DataSource.lookup(self._db, DataSource.LIBRARY_STAFF)
+
+        if flask.request.method == "GET":
+            lists = []
+            for entry in work.custom_list_entries:
+                list = entry.customlist
+                lists.append(dict(id=list.id, name=list.name))
+            return dict(custom_lists=lists)
+
+        if flask.request.method == "POST":
+            lists = flask.request.form.get("lists")
+            if lists:
+                lists = json.loads(lists)
+            else:
+                lists = []
+
+            # Remove entries for lists that were not in the submitted form.
+            submitted_ids = [l.get("id") for l in lists if l.get("id")]
+            for entry in work.custom_list_entries:
+                if entry.list_id not in submitted_ids:
+                    list = entry.customlist
+                    list.remove_entry(work)
+                    for lane in Lane.affected_by_customlist(list):
+                        lane.update_size(self._db)
```

Comment:
There's a chance this could cause a big performance hit. I think it would be better to keep a set of affected lanes and update their size outside of this loop.